torchdynamo/variables/torch.py

```diff
@@ -158,6 +158,13 @@ def call_function(
             and args[0].size is not None
         ):
             return ConstantVariable(product(args[0].size), **options)
+        elif self.value is torch._shape_as_tensor:
+            assert len(args) == 1
+            assert isinstance(args[0], TensorVariable)
+            log.warning(
+                "torch._shape_as_tensor(x) is going to be rewritten as x.size()"
```

Comment:
huh, is warning the right thing to do here? If you pass in the wrong number of args to `_shape_as_tensor()` we would throw an error in eager.

Reply:
I assert that the args[0] is a Tensor.